store/subscription.ts

```diff
@@ -0,0 +1,39 @@
+import Vue from 'vue'
+
+declare type Unsubscribe = () => void
+
+export interface SubscriptionStruct {
+  balanceSub: Unsubscribe
+}
+
+const defaultState: SubscriptionStruct = {
+  balanceSub: () => void 0,
+}
+
+// Disabling namespace to match with the original repo
+export const namespaced = false
+
+export const state = () => defaultState
+
+export const mutations = {
+  addSubscription(state: SubscriptionStruct, { key, sub }): void {
+    // unsubscribe previous subsription
+    const oldSub = state[key]
+    oldSub()
```

Comment:
@vikiival we are unsubscribing before any new subscription here.

Reply:
Oh ok